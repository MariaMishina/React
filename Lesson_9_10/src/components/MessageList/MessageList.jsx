import PropTypes from 'prop-types'

export function MessageList ({messages}) {
  return (
    <>
      <h1>ЧАТ</h1>
      <ul>
        {messages.map((message, index) => (
          <li key={index}>
            {message.author} : {message.text}
          </li>
        ))}
      </ul>
    </>
  )
}

MessageList.propTypes = {
  messages: PropTypes.array
}