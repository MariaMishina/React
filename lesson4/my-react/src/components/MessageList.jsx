import React from 'react';
import Message from './Message';

const MessageList = ({posts}) => {
    return (
        <div>
            {
              posts.map((post,) =>
                <div>
                  <Message post = {post} key = {post.id}/>
                </div>
              )
            }
        </div>
    )
}
export default MessageList;