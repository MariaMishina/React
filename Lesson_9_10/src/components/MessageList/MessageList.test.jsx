import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import {MessageList} from './MessageList'

describe('testing MessageList', () => {

    it('render component MessageList', () => {
        render(<MessageList messages={[]}/>)
      })
      it('render with snapshot', () => {
        const {asFragment} = render(<MessageList messages={[]}/>)
        expect(asFragment()).toMatchSnapshot()
       })
})