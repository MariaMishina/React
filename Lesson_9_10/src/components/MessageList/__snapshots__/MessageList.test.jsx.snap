// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing MessageList render with snapshot 1`] = `
<DocumentFragment>
  <h1>
    ЧАТ
  </h1>
  <ul />
</DocumentFragment>
`;
